Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPITest.mm

```diff
@@ -218,6 +218,42 @@
     EXPECT_NS_EQUAL(firstMessage, @{ @"key": @"One" });
 }
 
+TEST(WKWebExtensionAPITest, SendMessageBeforeEventListenerAdded)
+{
+    TestWebKitAPI::HTTPServer server({
+        { "/"_s, { { { "Content-Type"_s, "text/html"_s } }, ""_s } }
+    }, TestWebKitAPI::HTTPServer::Protocol::Http);
+
+    auto testMessage = @"Queued test message";
+
+    auto *contentScript = Util::constructScript(@[
+        @"browser.test.onMessage.addListener((message, data) => {",
+        [NSString stringWithFormat:@"  if (message != '%@') { return }", testMessage],
```

Comment:
```suggestion
        [NSString stringWithFormat:@"  if (message != '%@') return", testMessage],
```

Reply:
Maybe this should just `browser.test.assertEq(message, '%@')` since it should be the only message?